wandb/sdk/lib/retry.py

```diff
@@ -211,3 +214,76 @@ def wrapped_fn(*args: Any, **kargs: Any) -> Any:
         return wrapped_fn  # type: ignore
 
     return decorator
+
+
+class Backoff(abc.ABC):
+    """A backoff strategy: decides whether to sleep or give up when an exception is raised."""
+
+    @abc.abstractmethod
+    def next_sleep_or_reraise(self, exc: Exception) -> datetime.timedelta:
+        pass
```

Comment:
minor: maybe better to replace it with `raise NotImplementedError`?

Reply:
Yeah, sure! Fixed.